zk/conn.go

```diff
@@ -143,6 +150,9 @@ func ConnectWithDialer(servers []string, sessionTimeout time.Duration, dialer Di
 	if dialer == nil {
 		dialer = net.DialTimeout
 	}
+
+	newLogger := log.New(os.Stderr, "", log.LstdFlags)
```

Comment:
Absolutely. I should have gone with that pattern in the first place. I should go further and export the type. That way, someone could fall back to the original logger, if needed, without needing to dive in to the source to see how we create it.